src/test_l1_dsAudio.c

```diff
@@ -9954,6 +9954,117 @@ void test_l1_dsAudio_negative_dsGetHDMIARCPortId(void) {
 	UT_LOG("\n Out %s\n", __FUNCTION__);
 }
 
+/**
+* @brief Ensure dsSetAudioMixerLevels() sets the audio mixer levels of primary/system input
+*
+* **Test Group ID:** Basic: 01@n
+* **Test Case ID:** 163@n
+*
+* **Dependencies:** None@n
+* **User Interaction:** None
+*
+* **Test Procedure:**@n
+* |Variation / Step|Description|Test Data|Expected Result|Notes|
+* |:--:|-----------|----------|--------------|-----|
+* |01|Initialize Audio Ports using dsAudioPortInit() | | dsERR_NONE | Should Pass |
+* |02|Get the audio port handle for valid audio port type and valid index | type, index = [Loop through kPorts] , handle = [valid handle] | dsERR_NONE | Valid port handle must be returned |
+* |03|Call dsSetAudioMixerLevels() by looping through the primary audio inputs | handle = [valid handle], aInput = dsAudioInputPrimary , volume: 75 | dsERR_NONE | Should Pass |
+* |04|Terminate the Audio Port using dsAudioPortTerm() | | dsERR_NONE | Clean up after test |
+*
+*/
+void test_l1_dsAudio_positive_dsSetAudioMixerLevels(void) {
+	// Start of test
+	gTestID = 163;
+	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+	int result;
+	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+
+	// Step 01: Initialize Audio Port using dsAudioPortInit()
+	result = dsAudioPortInit();
+	UT_ASSERT_EQUAL(result, dsERR_NONE);
+
+	// Step 02: Get the port handle for all supported audio ports
+	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+		DS_ASSERT_AUTO_TERM_NUMERICAL(result, dsERR_NONE);
+		DS_ASSERT_AUTO_TERM_NOT_EQUAL(handle[i], null_handle);
+
+		// Step 03: Call dsSetAudioMixerLevels() for each port by looping through dsAudioInput_t enum
+		for (dsAudioInput_t audioInput = dsAudioInputPrimary; audioInput < dsAudioInputMax; audioInput++) {
+			DS_ASSERT_AUTO_TERM_NUMERICAL(dsSetAudioMixerLevels(handle[i], audioInput, volume), dsERR_NONE);
```

Comment:
please break this up to use UT_ASSERT Functions correctly, and not macro the offical macros

Reply:
how can you validate the that volume is set correct?